### PR TITLE
Add unit tests, VR tests, migration guide and readme for SpinButton

### PR DESCRIPTION
**Pull request by spmonahan**
## Current Behavior

`react-spinbutton` lacks most unit tests and visual regression (VR) tests.

## New Behavior

1. Adds unit tests
2. Adds VR tests
3. Adds Migration Guide
4. Adds README
5. Minor updates to SpinButton based on bugs uncovered in testing

## Related Issue(s)

#22307

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/cbf9e71dd01eb76d51a50f37981ab2c12f30a0f2'>cbf9e71dd01eb76d51a50f37981ab2c12f30a0f2</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-spinbutton/src/components/SpinButton/SpinButton.test.tsx [hunk 1]
```diff
@@ -11,10 +17,488 @@ describe('SpinButton', () => {
...
+    fireEvent.keyDown(spinButton, { key: Keys.Escape });
+
+    expect(spinButton.value).toEqual('1');
+    expect(onChange).not.toHaveBeenCalled();
+
+    spinButton.blur();
+    expect(onChange).not.toHaveBeenCalled();
+  });
+
+  it('is disabled', () => {
```
Comment (khmakoto) [inline packages/react-spinbutton/src/components/SpinButton/SpinButton.test.tsx:425]: _nit:_

```suggestion
  it('can be disabled', () => {
```

**Review by khmakoto — APPROVED** [review]
Approving with very minor comments

**Comment by khmakoto** [comment]
There seems to be a problem with `mouseDown` scenarios of min and max bound vr tests.

**Comment by spmonahan** [comment]
> There seems to be a problem with `mouseDown` scenarios of min and max bound vr tests.

Good catch! I've pushed some changes that should address this.

File: packages/react-spinbutton/src/components/SpinButton/useSpinButtonStyles.ts [hunk 2]
```diff
@@ -48,16 +48,20 @@ const useRootStyles = makeStyles({
...
+      right: 0,
+      bottom: 0,
+      left: 0,
       ...shorthands.borderRadius(tokens.borderRadiusMedium),
       pointerEvents: 'none',
       zIndex: 10,
     },
 
-    '::after': {
+    // TODO: change this to `::after`. Needs to be changed at the same time as react-input.
```
[the remark this answers is not in the document]
Comment (spmonahan) [reply packages/react-spinbutton/src/components/SpinButton/useSpinButtonStyles.ts:60]: Added #22365

**Review by raghavthind2005 — APPROVED** [review]

**Pull request merged**